lib/BackgroundJob/WatchActiveUsers.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Whiteboard\BackgroundJob;
+
+use OCA\Whiteboard\Service\ConfigService;
+use OCA\Whiteboard\Service\StatsService;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\TimedJob;
+
+class WatchActiveUsers extends TimedJob {
+	public function __construct(
+		ITimeFactory $time,
+		protected bool $isCLI,
+		protected StatsService $statsService,
+		protected ConfigService $configService,
+	) {
+		parent::__construct($time);
+		$this->setInterval(300);
+	}
+
+	protected function run($argument) {
+		if (!$this->configService->getWhiteboardEnableStatistics()) {
+			return;
+		}
+		$metricsData = $this->getMetricsData();
+		$activeUsers = $metricsData['totalUsers'] ?? 0;
+		$this->statsService->insertActiveUsersCount($activeUsers);
+	}
+
+	private function getMetricsData(): array {
+		$serverUrl = $this->configService->getCollabBackendUrl();
+		$metricToken = $this->configService->getCollabBackendMetricsToken();
+
+		if (!$serverUrl || !$metricToken) {
+			return [];
+		}
+
+		$curl = curl_init();
```

Comment:
Let's use the Nextcloud way of sending HTTP requests with https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/http_client.html